drivers/hyperv/hyperv.go

```diff
@@ -205,21 +210,32 @@ func (d *Driver) Create() error {
 		return err
 	}
 
-	log.Infof("Creating VM...")
-	virtualSwitch, err := d.chooseVirtualSwitch()
-	if err != nil {
-		return err
-	}
-
-	log.Infof("Using switch %q", virtualSwitch)
-
-	if err := cmd("Hyper-V\\New-VM",
+	args := []string{
+		"Hyper-V\\New-VM",
 		d.MachineName,
 		"-Path", fmt.Sprintf("'%s'", d.ResolveStorePath(".")),
-		"-SwitchName", quote(virtualSwitch),
-		"-MemoryStartupBytes", toMb(d.Memory)); err != nil {
+		"-MemoryStartupBytes", toMb(d.Memory),
+	}
+	if d.VirtualSwitch != "" {
+		log.Infof("Creating VM...")
+		virtualSwitch, err := d.chooseVirtualSwitch()
+		if err != nil {
+			return err
+		}
+		log.Infof("Using switch %q", virtualSwitch)
+		args = append(args, "-SwitchName", quote(virtualSwitch))
+	}
+
+	if err := cmd(args...); err != nil {
 		return err
 	}
+
+	if d.VirtualSwitch == "" {
```

Comment:
can't this move up and be part of the else clause of:
```go
if d.VirtualSwitch != "" {
```

Reply:
VM is created line 229 here in `cmd(args...)`. `else` is before and too early in the process.